File: Day 20/day20.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int n;
    cin >> n;
    int numSwaps=0;
    vector<int> a(n);
    for(int a_i = 0; a_i < n; a_i++){
    	cin >> a[a_i];
    }
    // Write Your Code Here
    for (int i = 0; i < n; i++) {
    // Track number of elements swapped during a single array traversal
    int numberOfSwaps = 0;

    for (int j = 0; j < n - 1; j++) {
        // Swap adjacent elements if they are in decreasing order
        if (a[j] > a[j + 1]) {
            swap(a[j], a[j + 1]);
            numberOfSwaps++;
            numSwaps++;
        }
    }

    // If no elements were swapped during a traversal, array is sorted
    if (numberOfSwaps == 0) {
        break;
    }
}
cout<<"Array is sorted in "<<numSwaps<<" swaps."<<endl;
cout<<"First Element: "<<a[0]<<endl;
cout<<"Last Element: "<<a[n-1];
    return 0;
}

File: Day 20/day20.py
#!/bin/python3

import sys

n = int(input().strip())
a = list(map(int, input().strip().split(' ')))
# Write Your Code Here
numSwap = 0
for num in range(len(a)-1, 0, -1):
    for i in range(num):
        if a[i] > a[i+1]:
            a[i], a[i+1] = a[i+1], a[i]
            numSwap +=1


print("Array is sorted in {} swaps.".format(numSwap))
print("First Element: {}".format(a[0]))
print("Last Element: {}".format(a[-1]))
